easybuild/easyconfigs/h/HEASoft/HEASoft-6.27.2-GCCcore-4.9.3.eb

```diff
@@ -0,0 +1,89 @@
+name = 'HEASoft'
+version = '6.27.2'
+
+
+dependencies = [
+    ( 'CFITSIO', '3.48' ),
+    ( 'Perl', '5.30.2' ),
+]
+
+easyblock = 'ConfigureMake'
+
+homepage = ''
+description = """"""
+
+toolchain = {'name': 'GCCcore', 'version': '4.9.3'}
+toolchainopts = {'pic': True}
+
+source_urls = [
+    'https://www.isdc.unige.ch/~savchenk/cache/heasoft/',
+    'https://heasarc.gsfc.nasa.gov/FTP/software/lheasoft/lheasoft%(version)s/',
+]
+sources = ['heasoft-%(version)ssrc_no_xspec_modeldata.tar.gz']
+patches = []
+
+preconfigopts = "cd BUILD_DIR/ &&"
+prebuildopts = "cd BUILD_DIR/ &&"
+
+# build fails with parallel
+parallel = False
+
+preinstallopts = "cd BUILD_DIR/ &&"
+
+# note that "x86_64-pc-linux-gnu-libc2.17" is hardcoded, since it is derived from toolchain and platform
+# it might differ in different case: to be checked
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['x86_64-pc-linux-gnu-libc2.17/bin', 'x86_64-pc-linux-gnu-libc2.17/lib'],
+}
+
+modextrapaths = {
+    'PATH': 'x86_64-pc-linux-gnu-libc2.17/bin',
+    'LD_LIBRARY_PATH': 'x86_64-pc-linux-gnu-libc2.17/lib',
+}
+
+modextravars = dict(
+    HEADAS="%(installdir)s/x86_64-pc-linux-gnu-libc2.17",
+    EXT="lnx",
+    LYNX_CFG="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/lib",
+    FTOOLS="%(installdir)s/x86_64-pc-linux-gnu-libc2.17",
+    FTOOLSINPUT="stdin",
+    FTOOLSOUTPUT="stdout",
+    # TODO: how to set this? not critical in many cases
+    #LHEAPERL="......local/easybuild/software/Perl/5.30.2-GCCcore-4.9.3/bin/perl", 
+    LHEASOFT="%(installdir)s/x86_64-pc-linux-gnu-libc2.17",
+    LHEA_DATA="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/refdata",
+    LHEA_HELP="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/help",
+    PERL5LIB="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/lib/perl",
+    PERLLIB="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/lib/perl",
+    PFCLOBBER="1",
+    PGPLOT_DIR="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/lib",
+    PGPLOT_FONT="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/lib/grfont.dat",
+    PGPLOT_RGB="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/lib/rgb.txt",
+    POW_LIBRARY="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/lib/pow",
+    TCLRL_LIBDIR="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/lib",
+    XANADU="%(installdir)s",
+    XANBIN="%(installdir)s/x86_64-pc-linux-gnu-libc2.17",
+    XRDEFAULTS="%(installdir)s/x86_64-pc-linux-gnu-libc2.17/xrdefaults",
+)
+
+modluafooter = """
+local home    = os.getenv("HOME")
+pushenv("PFILES", pathJoin(home,"pfiles") .. ";%(installdir)s/x86_64-pc-linux-gnu-libc2.17/syspfiles")
```

Comment:
This seems like it would be solved by having just modextrapaths extend `PFILES` in this easyconfig, which will append to the path variable instead of replacing it (which is what modextravars does).
This method with a footer will not work well, because unloading the module will not restore it (and a bunch of other problems like this)

Reply:
Is it possible to get the user home path?

Here I really need to replace `PFILES`, not append. So I would put it to `modextravars`.
But maybe there is a way to use user home in the module?

Reply:
> This method with a footer will not work well, because unloading the module will not restore it (and a bunch of other problems like this)

Concerning unloading, I thought `pushenv` (unlike `setenv`) will preserve previous variable state and restore it on unload, according to the [doc](https://lmod.readthedocs.io/en/latest/050_lua_modulefiles.html):

```
pushenv (“NAME”, “value”):
    sets NAME to value just like setenv. In addition it saves the previous value in a hidden environment variable. This way the previous state can be returned when a module is unloaded. pushenv (“FOO”,false) will clear “FOO” and the pop will return the previous value.
```
(there is also a [tcl version](https://lmod.readthedocs.io/en/latest/051_tcl_modulefiles.html))

I tried it and it seems to work: the variable is restored to its pre-load value with unload. 

But I am not really experienced with modules and easybuild. I also note that no other easyconfig seems to use pushenv.
So I would be happy to use a better, more accepted, solution if there is one. 
I just do not know how else to set variable including dynamic `$HOME` path.